src/EnergyPlus/api/datatransfer.cc

```diff
@@ -316,60 +392,120 @@ int getPluginTrendVariableHandle(const char* name) {
 
 Real64 getPluginTrendVariableValue(int handle, int timeIndex) {
     if (handle < 0 || handle > EnergyPlus::PluginManagement::pluginManager->maxTrendVariableIndex) {
-        throw std::runtime_error("Plugin Trend Variable index out of range");
+        // need to fatal out once the plugin is done
+        // throw an error, set the fatal flag, and then return
+        EnergyPlus::ShowSevereError("Data Exchange API: Problem -- index error in getPluginTrendVariableValue; received handle: " + std::to_string(handle));
+        EnergyPlus::ShowContinueError("The getPluginTrendVariableValue function will return 0 to allow the plugin to finish, then EnergyPlus will abort");
+        EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        return 0;
     }
     if (timeIndex < 2 || timeIndex > ((int)EnergyPlus::PluginManagement::pluginManager->getTrendVariableHistorySize(handle) - 1)) {
```

Comment:
The trend variable is just a wrapper over a regular plugin variable.  The plugin variable must be first declared and assigned, then the trend is used to track the value of it.  Thus the plugin should already have access to the underlying variable, and would just grab the current value of that variable directly rather than doing a trend lookup.  I guess I could just make this go down to one though, it doesn't hurt anything.